File: 1d_json/python/person.py
import json


class Person:
    def __init__(self, name, age, job, verified, parents):
        self.name = name
        self.age = age
        self.job = job
        self.verified = verified
        self.parents = parents

    def __str__(self):
        return ", ".join([f"{k}: {v}" for k, v in self.__dict__.items()])


class MyEncoder(json.JSONEncoder):
    def default(self, o):
        return o.__dict__


class MyDecoder(json.JSONDecoder):
    def decode(self, s):
        d = json.JSONDecoder.decode(self, s)
        return Person(**d)


if __name__ == '__main__':
    bob = Person(name="Bob", age=12, job=None, verified=True,
            parents=["Alice", "Carl"])
    bob_json = json.dumps(bob, cls=MyEncoder)
    print(bob_json)
    bob = json.loads(bob_json, cls=MyDecoder)
    print(bob)

File: 1d_json/rust/json/src/bin/person.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize, Serialize)]
struct Person {
    name: String,
    age: usize,
    job: Option<String>,
    verified: bool,
    parents: Vec<String>,
}

fn main() {
    let bob = Person {
        name: "Bob".to_string(),
        age: 12,
        verified: true,
        job: None,
        parents: ["Alice".to_string(), "Carl".to_string()].to_vec(),
    };
    let json = serde_json::to_string(&bob).unwrap();
    println!("{}", &json);

    let person: Person = serde_json::from_str(&json).unwrap();

    println!("{:?}", person);
}
